docs/design/kuadrant-api-v2.md

```diff
@@ -0,0 +1,178 @@
+kuadrant control plane, having as input the API Product resource.
+
+For example, cluster operators can deploy required (cloud provider level) networking resources to
+forward traffic to the cluster ingress gateways. Then, the API owners can create (cluster level)
+networking resources, like kubernetes Ingress or Gateway API HttpRoutes, to route that traffic to
+the kuadrant gateways. In this example only *North/South* traffic was referenced. By design,ç
+it could also be *East/west* traffic. By design, there is nothing preventing from having *East/West*
+traffic being protected by kuadrant. This is mainly because kuadrant is not a networking API.
+
+**Protected API specification**
+
+Kuadrant will support the open standard
+[OpenAPI 3.X specification](https://github.com/OAI/OpenAPI-Specification/blob/main/versions/3.0.2.md)
+for the API spec.
+
+Each *API Product* can hold multiple OAS docs references. The *API Product* `spec` field includes references
+in the form of URLs and config map references. The actual content may be included in the `status` field.
+
+Kuadrant will extend the OAS doc leveraging
+[OAS custom extensions](https://github.com/OAI/OpenAPI-Specification/blob/main/versions/3.0.2.md#specificationExtensions)
+`x-kuadrant-*`, initially for rate limit purposes. In the future, OAS extensions could be leveraged
+for other API management features, for instance usage plans. More about the kuadrant custom
+extensions in the spec reference section.
+
+In case of lacking one OpenAPI doc, kuadrant will provide tooling for easy generation of OpenAPI docs.
+
+Any kuadrant configuration at API path or operation level will be defined as OpenAPI kuadrant extension.
+
+**Backends - Upstreams**
+
+The protected API source, owned by the kuadrant customer, can be specified in two forms:
+* A (base) URL
+* A kubernetes service reference (name and namespace).
+
+When the source is a kubernetes service, kuadrant provides annotations and labels that can be added
+to the service. These annotations and labels can specify, for example, the OAS reference.
+These annotations and labels aim to make the `spec` of the *API Product* resource simpler to specify.
+
+**AuthConfig Templates**
+
+The configuration of the API authentication requires *three* steps. Only the first
+two steps involve kuadrant, but all three of them are required to have the system working.
+For completeness, all of them are listed here.
+
+1. Operation based authentication config
+
+The customer needs to specify which API operation (method + path) requires authentication and
+which type of authentication (API key, oauth2, oidc, http basic, etc). OpenAPI spec fully covers
+this use case and kuadrant can read this configuration from OAS doc. For any authN type not covered
+by OAS 3.X, kuadrant could provide extensions for added authentication types. OpenAPI 3.x does
+not support authentication requirements at path level, only global or operation level.
```

Comment:
I found this paragraph a bit unclear. Initially seemed OAS supported a path based authentication but then the last sentence seems to state it doesnt?

Reply:
OpenAPI 3.x supports operation level auth requirements. however, do not ask me why, you cannot say "I want to apply this auth requirement to `/toystore/dolls` for all HTTP methods". Instead you need to duplicate the auth requirement in all HTTP methods for that path.

I will rewrite the paragraph

Reply:
The link between a path and a security scheme (e.g. authn) can be through an operation and its "security requirements":
- pathA → operation1 → securityRequirement → securityScheme_x
- pathB → operation1 → securityRequirement → securityScheme_x
- pathC → operation2 → securityRequirement → securityScheme_x
- pathD → operation3 → securityRequirement → securityScheme_y

In the above, pathA, pathB and pathC all share the same authentication scheme, whereas pathD can have its own.